### PR TITLE
fix building with clang 9

### PR DESCRIPTION
**Pull request by mytbk**
There seems to be some API change in the new clang version.
Fix by copying the code from the clang source code.

### CODE REVIEW

**Comment by mytbk** [comment]
This may break building with previous clang. I have not tested with the old systems yet.

File: src/lib_cxx/data/parser/cxx/GeneratePCHAction.cpp [hunk 1]
```diff
@@ -36,14 +37,11 @@ std::unique_ptr<clang::ASTConsumer> GeneratePCHAction::CreateASTConsumer(
...
-		CI.getPreprocessor(),
-		OutputFile,
-		Sysroot,
-		Buffer,
+	Consumers.push_back(std::make_unique<clang::PCHGenerator>(
+		CI.getPreprocessor(), CI.getModuleCache(), OutputFile, Sysroot, Buffer,
 		FrontendOpts.ModuleFileExtensions,
-		true,
-		FrontendOpts.IncludeTimestamps));
+		CI.getPreprocessorOpts().AllowPCHWithCompilerErrors,
```
Comment (mytbk) [inline src/lib_cxx/data/parser/cxx/GeneratePCHAction.cpp]: Oh, just a copy-paste from the clang code, should this still be "true" here?

**Comment by egraether** [comment]
Thanks for looking into this. Switching the Clang version is always a bit of work, so we don't want to surprise contributors with this change. It needs to be properly communicated. We are not well organized regarding contributor collaboration yet. We scheduled the official switch for Nov 25th 2019. Please see #763

**Comment by mlangkabel** [comment]
In order to get this running on Windows, I need to add `riscvasmparser riscvcodegen` to the list of linked llvm libraries in `llvm_map_components_to_libnames(...)`

File: src/lib_cxx/data/parser/cxx/GeneratePCHAction.cpp (hunk 1 above)
Comment (mlangkabel) [reply src/lib_cxx/data/parser/cxx/GeneratePCHAction.cpp]: yes

**Comment by mytbk** [comment]
> In order to get this running on Windows, I need to add `riscvasmparser riscvcodegen` to the list of linked llvm libraries in `llvm_map_components_to_libnames(...)`

The LLVM RISC-V support is not related to clang. I think it's because of the LLVM configuration.

> 
> Edit: In the main CMakeLists file, I mean.

File: src/lib_cxx/data/parser/cxx/GeneratePCHAction.cpp (hunk 1 above)
Comment (mytbk) [reply src/lib_cxx/data/parser/cxx/GeneratePCHAction.cpp]: changed

**Comment by mlangkabel** [comment]
Hm, if I set up llvm and clang as proposed [here](https://clang.llvm.org/get_started.html), I get the linker error that risk-v related symbols are not found (sing the msvc compiler). This will probably be an issue for everyone having clang setup this way, right?

**Comment by mlangkabel** [comment]
It looked like we needed the changes from my previous comment. So we pulled #799 instead. However, if you have a suggestion for avoiding `riscvasmparser` and `riscvcodegen`, please let us know.

**Pull request closed without merging**